dialdbg/src/parse.rs

```diff
@@ -0,0 +1,326 @@
+        if DEVELOPMENT.is_some() {
+            writeln!(out, "log message: {log}")?;
+        }
+
+        if log.contains(strings::DIAL_ERROR_PREFIX) {
+            res.dial_error_message = Some(extract_dial_error(log)?);
+        } else if log.contains(strings::MDNS_ADDRESS_FOUND) {
+            res.mdns_address = Some(extract_mdns_address(log)?);
+        } else if log.contains(strings::MDNS_QUERY_ATTEMPT) {
+            mdns_query_start = Some(extract_timestamp(log)?);
+        } else if log.contains(strings::MDNS_QUERY_SUCCESS) {
+            match mdns_query_start {
+                Some(mqs) => {
+                    res.mdns = Some(extract_timestamp(log)?.signed_duration_since(mqs));
+                }
+                None => {
+                    return Err(anyhow!(
+                        "expected '{}' log before '{}'",
+                        strings::MDNS_QUERY_ATTEMPT,
+                        strings::MDNS_QUERY_SUCCESS
+                    ));
+                }
+            }
+        } else if log.contains(strings::ACQUIRING_AUTH_TOKEN) {
+            authentication_start = Some(extract_timestamp(log)?);
+        } else if log.contains(strings::ACQUIRED_AUTH_TOKEN) {
+            match authentication_start {
+                Some(aus) => {
+                    res.authentication = Some(extract_timestamp(log)?.signed_duration_since(aus));
+                }
+                None => {
+                    return Err(anyhow!(
+                        "expected '{}' log before '{}'",
+                        strings::ACQUIRING_AUTH_TOKEN,
+                        strings::ACQUIRED_AUTH_TOKEN
+                    ));
+                }
+            }
+        } else if log.contains(strings::DIAL_ATTEMPT) {
+            connection_establishment_start = Some(extract_timestamp(log)?);
+        } else if log.contains(strings::DIALED_GRPC) {
+            match connection_establishment_start {
+                Some(ces) => {
+                    res.connection = Some(extract_timestamp(log)?.signed_duration_since(ces));
+                }
+                None => {
+                    return Err(anyhow!(
+                        "expected '{}' log before '{}'",
+                        strings::DIAL_ATTEMPT,
+                        strings::DIALED_GRPC
+                    ));
+                }
+            }
+        }
+    }
+
+    write!(out, "{res}")?;
+    Ok(())
+}
+
+fn extract_ice_candidate_pair(log: &str) -> Result<String> {
+    // Tear off LOG prefixes.
+    let split_log = log
+        .split(strings::CANDIDATE_SELECTED)
+        .collect::<Vec<&str>>();
+    if split_log.len() != 2 {
+        return Err(anyhow!("malformed selected candidate message: {log}"));
+    }
+
+    // Remove annoying ": " still left over from log.
+    Ok(split_log[1].strip_prefix(": ").unwrap().to_string())
+}
+
+pub(crate) fn parse_webrtc_logs(log_path: PathBuf, out: &mut Box<dyn io::Write>) -> Result<()> {
```

Comment:
[opt] This function is surprisingly named. I would have expected it to return `anyhow::Result<WebRTCResult>` given its current name. Maybe something like `write_webrtc_log_statistics`? 

Similar with other functions in this file. But probably not important

Reply:
Good point; Modified them to return `Result<*Result>` and actually printed the structs in `main.rs`.